include/spdlog/details/circular_q.h

```diff
@@ -72,6 +72,20 @@ class circular_q
         return v_[head_];
     }
 
+    // Return number of elements actually stored
+    size_t size() const
+    {
+        return (tail_ - head_) % max_items_;
```

Comment:
I think it breaks when head_ > tail_ (can happen)

Reply:
Sorry, a stupid mistake. Should be solved now.

Reply:
no need to ``` % max_items``` and will give wrong result if if tail_ == head_(i.e. if empty it will give max_items_) 

Here is a fixed version:
```c++
size_t size() const
    {
        if (tail_ >= head_)
        {
           return (tail_ - head_);
        }
        else
        {
            return max_items_ - (head_ - tail_ );
        }
    }
```